APIs/schemas/source_data.json

```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "description": "Describes a data Source",
+  "title": "Data Source resource",
+  "allOf": [
+    { "$ref": "source_core.json" },
+    {
+      "type": "object",
+      "required": [
+        "format"
+      ],
+      "properties": {
+        "format": {
+          "description": "Format of the data coming from the Source as a URN",
+          "type": "string",
+          "enum": [
+            "urn:x-nmos:format:data"
+          ],
+          "format": "uri"
+        },
```

Comment:
One thing I like about IS-04 classes (unlike IS-05 transport parameters :( ) is that the type of the polymorphic objects can be deduced from a single field, e.g. "format": "urn:x-nmos:format:audio" would indicate that it is an audio flow or source. That is extremely useful when deserializing in typed objects in a generic way. 
Unfortunately, this breaks that concept as there is an overlap in what "format": "urn:x-nmos:format:data" designates (it could be either source_generic.json or source_data.json). 
There would be a workaround for IS-07 sources as they would have "event_type" and one could look for its existence, but it doesn't cover the most general case. (I assume "event_type" is not required to make this extensible to other specs that might use "urn:x-nmos:format:data"). But this relates to my other comment on this PR.

Reply:
I don't believe it can be source_generic or source_data, as the URN was removed from source_generic, or am I misunderstanding and you'd prefer it remained a 'subclass' of that type? The same can be said for the existing source_audio.

Reply:
I also brought up the point of data-binding JSON bodies to polymorphic types. However, it's not entirely possible already, hence a few instances of `"anyOf"` rather than `"oneOf"` in the schemas already. For what it's worth, I think the IS-05 v1.1 `/transporttype` solves your problem there (albeit in a different endpoint).

I don't think there is an overlap for `"format": "urn:x-nmos:format:data"` though? That is source_data.json not source_generic.json, according to this PR.

I made a related comment as to the applicability of `"event_type"` to other specs here - https://github.com/bbc/nmos-discovery-registration/pull/7#issuecomment-498330172.

Reply:
I apologize, didn't notice the removal of data from source_generic.json